include/geometrycentral/surface/embedded_geometry_interface.h

```diff
@@ -84,6 +135,96 @@ class EmbeddedGeometryInterface : public ExtrinsicGeometryInterface {
   virtual void computeHalfedgeCotanWeights() override;
   virtual void computeEdgeCotanWeights() override;
+
+  // == Polygon Operators
+
+  // = Bunge et al. "Polygon Laplacian Made Simple" (2020), based on virtual refinement (virtual node method).
+  // Copyright (C) 2020 Astrid Bunge, Philipp Herholz, Misha Kazhdan, Mario Botsch, MIT license
+  // (Modified to work in geometry-central. Original code can be found here:
+  // https://github.com/mbotsch/polygon-laplacian)
+
+  // Laplacian
+  DependentQuantityD<Eigen::SparseMatrix<double>> simplePolygonLaplacianQ;
+  virtual void computeSimplePolygonLaplacian();
+
+  // Vertex mass matrix (unlumped)
+  DependentQuantityD<Eigen::SparseMatrix<double>> simplePolygonVertexGalerkinMassMatrixQ;
+  virtual void computeSimplePolygonVertexGalerkinMassMatrix();
+
+  // Vertex mass matrix (lumped)
+  DependentQuantityD<Eigen::SparseMatrix<double>> simplePolygonVertexLumpedMassMatrixQ;
+  virtual void computeSimplePolygonVertexLumpedMassMatrix();
+
+  // helper functions
+  FaceData<Eigen::VectorXd> virtualRefinementAreaWeights;
+  DependentQuantityD<FaceData<Eigen::VectorXd>> virtualRefinementAreaWeightsQ; // affine weights for each virtual node
+  virtual void computeVirtualRefinementAreaWeights();
+  virtual Eigen::MatrixXd simplePolygonMassMatrix(const Face& f);
+  virtual Eigen::MatrixXd simplePolygonStiffnessMatrix(const Face& f);
+  virtual SparseMatrix<double> simplePolygonProlongationMatrix();
+  virtual Eigen::MatrixXd polygonPositionMatrix(const Face& f);
+  virtual Eigen::VectorXd simplePolygonVirtualVertex(const Eigen::MatrixXd& poly) const;
+
+  // = de Goes et al. "Discrete Differential Operators on Polygonal Meshes" (2020), based on the virtual element method.
+  // Use of this source code is governed by a LGPL-3.0 license.
+  // (Modified to work in geometry-central. Original code can be found here:
+  // https://github.com/DGtal-team/DGtal/blob/master/src/DGtal/dec/PolygonalCalculus.h)
+
+  // Laplacian
+  DependentQuantityD<Eigen::SparseMatrix<double>> polygonLaplacianQ;
+  virtual void computePolygonLaplacian();
+
+  DependentQuantityD<Eigen::SparseMatrix<double>> polygonGradientMatrixQ;
+  virtual void computePolygonGradientMatrix();
+
+  DependentQuantityD<Eigen::SparseMatrix<double>> polygonDivergenceMatrixQ;
+  virtual void computePolygonDivergenceMatrix();
+
+  // Vertex mass matrix (lumped)
+  DependentQuantityD<Eigen::SparseMatrix<double>> polygonVertexLumpedMassMatrixQ;
+  virtual void computePolygonVertexLumpedMassMatrix();
+
+  // Vertex connection Laplacian
+  DependentQuantityD<Eigen::SparseMatrix<std::complex<double>>> polygonVertexConnectionLaplacianQ;
+  virtual void computePolygonVertexConnectionLaplacian();
+
+  // DEC Operators
+  std::array<Eigen::SparseMatrix<double>*, 7> polygonDECOperatorArray;
+  DependentQuantityD<std::array<Eigen::SparseMatrix<double>*, 7>> polygonDECOperatorsQ;
+  virtual void computePolygonDECOperators();
+
+  // helper functions
```

Comment:
These are a _lot_ of helper functions. I'm worried this class/file is getting too busy.

Could we maybe move these helpers to a separate file, and invoke them as-needed? And maybe once moved to their own file they no-longer need to be virtual?